src/Coordination/Changelog.cpp

```diff
@@ -439,11 +480,12 @@ void Changelog::removeAllLogsAfter(uint64_t remove_after_log_start_index)
 
     size_t start_to_remove_from_log_id = start_to_remove_from_itr->first;
 
+    const auto timestamp_folder = changelogs_detached_dir / getCurrentTimestampFolder();
     /// All subsequent logs shouldn't exist. But they may exist if we crashed after writeAt started. Remove them.
     for (auto itr = start_to_remove_from_itr; itr != existing_changelogs.end();)
     {
-        LOG_WARNING(log, "Removing changelog {}, because it's goes after broken changelog entry", itr->second.path);
-        std::filesystem::remove(itr->second.path);
+        LOG_WARNING(log, "Removing changelog {}, because it goes after broken changelog entry", itr->second.path);
+        removeLog(itr->second.path, timestamp_folder);
```

Comment:
Consider moving `getCurrentTimestampFolder()` call and this `LOG_WARNING` inside `removeLog`

Reply:
I create timestamp folder so a single call to `removeAllLogs`/`removeAllLogsAfter` move all the broken logs to the same folder.
LOG_WARNING should be actually different, so I should actually modify the message in the `removeAllLogs`.